pkg/transform/kubernetes/populators.go

```diff
@@ -0,0 +1,164 @@
+package kubernetes
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/kedgeproject/kedge/pkg/spec"
+
+	"sort"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/pkg/errors"
+	api_v1 "k8s.io/client-go/pkg/api/v1"
+)
+
+func populateProbes(c spec.Container) (spec.Container, error) {
+	// check if health and liveness given together
+	if c.Health != nil && (c.ReadinessProbe != nil || c.LivenessProbe != nil) {
+		return c, fmt.Errorf("cannot define field 'health' and " +
+			"'livenessProbe' or 'readinessProbe' together")
+	}
+	if c.Health != nil {
+		c.LivenessProbe = c.Health
+		c.ReadinessProbe = c.Health
+		c.Health = nil
+	}
+	return c, nil
+}
+
+func searchConfigMap(cms []spec.ConfigMapMod, name string) (spec.ConfigMapMod, error) {
+	for _, cm := range cms {
+		if cm.Name == name {
+			return cm, nil
+		}
+	}
+	return spec.ConfigMapMod{}, fmt.Errorf("configMap %q not found", name)
+}
+
+func getSecretDataKeys(secrets []spec.SecretMod, name string) ([]string, error) {
+	var dataKeys []string
+	for _, secret := range secrets {
+		if secret.Name == name {
+			for dk := range secret.Data {
+				dataKeys = append(dataKeys, dk)
+			}
+			for sdk := range secret.StringData {
+				dataKeys = append(dataKeys, sdk)
+			}
+			return dataKeys, nil
```

Comment:
not sure if I should do that, because it is not what the function is supposed to be doing according to its name